src/engine/routing_algorithms/many_to_many.cpp

```diff
@@ -33,8 +33,7 @@ struct NodeBucket
     }
 };
 
-// FIXME This should be replaced by an std::unordered_multimap, though this needs benchmarking
-using SearchSpaceWithBuckets = std::unordered_map<NodeID, std::vector<NodeBucket>>;
+using SearchSpaceWithBuckets = std::unordered_multimap<NodeID, NodeBucket>;
```

Comment:
😱 https://bugs.llvm.org/show_bug.cgi?id=16747
```
clang-3.3 with libc++ from 2013-01-31: 
Duration for 123456 integer inserts: 21836ms

gcc-4.6 with GLIBCXX_3.4.16:
Duration for 123456 integer inserts: 7ms
```

thanks for pointing! But i will definitely  re-check with the current gcc/clang versions